src/org/labkey/test/components/react/MultiMenu.java

```diff
@@ -186,34 +206,182 @@ public List<String> getMenuText()
+        {
+            toggle.click(); // expand the toggle
+            WebDriverWrapper.waitFor(() ->
+                            Locator.tagWithClass("span", "fa-chevron-up").existsIn(toggle),
+                    "the toggle-item did not expand in time", 1000);
+            WebDriverWrapper.waitFor(()-> getListItems().size() > listItemCount,
+                    "the list-items did not appear in the list after expanding", 1000);
+        }
+
+        return toggle;
+    }
+
+    /**
+     * gets the names of dropdown-section__menu-items between a heading and the next divider
+     *
+     * @param heading
+     * @return
+     */
+    public List<String> getItemsUnderHeading(String heading, boolean collapse)
+    {
+        var itemTexts = getWrapper().getTexts(getItemsUnderHeading(heading));
+        if (collapse)
+            collapse();
+        return itemTexts;
+    }
+
+    /**
+     * gets the dropdown-section__menu-items between a heading and the next separator
+     *
+     * @param heading
+     * @return
+     */
+    protected List<WebElement> getItemsUnderHeading(String heading)
     {
         expandAll();
         boolean headingFound = false;
-        List<String> items = new ArrayList<>();
+        List<WebElement> itemsUnderHeading = new ArrayList<>();
         List<WebElement> listItems = Locator.tag("li").findElements(this);
 
         for (WebElement item : listItems)
         {
             String className = item.getAttribute("class");
+            String role = item.getAttribute("role");
             String text = item.getText().trim();
 
             if (className.contains("dropdown-header") && text.equalsIgnoreCase(heading))
                 headingFound = true;
 
-            // Once we've found our header we know that all dropdown-section__menu-item elements belong to the heading
+            // Once we've found our header we know that all presentation elements belong to the heading
             // we are interested in
-            if (headingFound && className.contains("dropdown-section__menu-item"))
-                items.add(text);
+            if (headingFound && role.equals("presentation"))
+                itemsUnderHeading.add(item);
 
             // Once we hit a divider we're done looking at menu items related to the heading, so we can stop iterating
-            if (headingFound && className.contains("divider"))
+            if (headingFound && role.equals("separator"))
                 break;
         }
 
-        collapse();
+        return itemsUnderHeading;
+    }
+
+    private WebElement getItemUnderHeading(String toggleLabel, String label)
+    {
+        var itemsUnderHeading = getItemsUnderHeading(toggleLabel);
+        for (WebElement item : itemsUnderHeading)
+        {
+            if (item.getText().trim().equals(label))
+                return item;
+        }
+        throw new NoSuchElementException("No item '" + label + "' under toggle '" + toggleLabel + "' found.");
+    }
```

Comment:
`toggle` ≠ `heading`, right?
```suggestion
    private WebElement getItemUnderHeading(String heading, String label)
    {
        var itemsUnderHeading = getItemsUnderHeading(heading);
        for (WebElement item : itemsUnderHeading)
        {
            if (item.getText().trim().equals(label))
                return item;
        }
        throw new NoSuchElementException("No item '" + label + "' under heading '" + heading + "' found.");
    }
```